iothub/service/tests/Amqp/AmqpConnectionHandlerTests.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using FluentAssertions;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Amqp;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Azure.Devices.Tests.Amqp
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class AmqpConnectionHandlerTests
+    {
+        private const string HostName = "contoso.azure-devices.net";
+        private static readonly IIotHubServiceRetryPolicy noRetryPolicy = new IotHubServiceNoRetry();
+        private static readonly IotHubServiceClientOptions s_options = new()
+        {
+            Protocol = IotHubTransportProtocol.Tcp,
+            RetryPolicy = noRetryPolicy
+        };
+        private static EventHandler ConnectionLossHandler = (object sender, EventArgs e) => { };
+
+        [TestMethod]
+        public async Task AmqpConnectionHandler_SendAsync()
+        {
+            // arrange
+            using var amqpMessage = AmqpMessage.Create();
+
+            var mockCredential = new Mock<TokenCredential>();
+            var tokenCredentialProperties = new IotHubTokenCredentialProperties(HostName, mockCredential.Object);
+            var mockAmqpConnection = new Mock<AmqpConnection>();
+            var mockWorkerSession = new Mock<AmqpSessionHandler>();
+            var mockAmqpCbsSessionHelper = new Mock<AmqpCbsSessionHandler>();
+
+            using var connectionHandler = new AmqpConnectionHandler(
+                tokenCredentialProperties,
+                IotHubTransportProtocol.Tcp,
+                HostName,
+                s_options,
+                ConnectionLossHandler,
+                mockAmqpCbsSessionHelper.Object,
+                mockWorkerSession.Object);
+
+            Outcome outcomeToReturn = new Accepted();
+
+            mockWorkerSession
+                .Setup(ws => ws.SendAsync(It.IsAny<AmqpMessage>(), It.IsAny<ArraySegment<byte>>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(outcomeToReturn);
+
+            var ct = new CancellationToken();
+
+            // act
+            Outcome act = await connectionHandler.SendAsync(amqpMessage, ct).ConfigureAwait(false);
+
+            // assert
+            act.Should().BeEquivalentTo(new Accepted());
```

Comment:
I'm not sure the value of this test/assertion. Can you elaborate?

Reply:
In AmqpConnectionHandler, the SendAsync method invokes the workerSession to send the message along a delivery tag and the outcome of this call is returned. This test is simulating the workerSession by having it return a "Accepted" outcome upon the SendAsync call, and the assertion confirms that the correct outcome is returned in the end.